common/djangoapps/third_party_auth/saml.py

```diff
@@ -84,6 +85,17 @@ def generate_saml_config(self, idp=None):
         else:
             return super().generate_saml_config()
 
+    def auth_complete(self, *args, **kwargs):
+        """
+        Handle exceptions that happen during SAML authentication
+        """
+        try:
+            return super().auth_complete(*args, **kwargs)
+        # We are seeing errors of MultiValueDictKeyError looking for the parameter 'RelayState'.
+        # We would like to have a more specific error to handle for observability purposes.
+        except MultiValueDictKeyError as e:
+            raise AuthMissingParameter(self.name, e.args[0])
```

Comment:
FYI: I saw example code (that could have been wrong), that just sent self as the first parameter. Just checking.

Also, it seems we have no unit test for this. Can you fix that?

Reply:
That was my original version of the code (which was wrong, sorry).